net/acme/files/acme.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/sh
+# Wrapper for acme.sh to work on openwrt.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# Authors: Toke Høiland-Jørgensen <toke@toke.dk>
+
+export state_dir='/etc/acme'
+export account_email=
+export debug=0
+export challenge_dir='/var/run/acme/challenge'
+IPTABLES_INSTALLED=0
+HOOK=/usr/lib/acme/hook
+LOG_TAG=acme
+
+. /lib/functions.sh
+. /usr/lib/acme/functions.sh
+
+cleanup() {
+	debug "cleaning up"
+	if [ "$IPTABLES_INSTALLED" = 1 ]; then
+		iptables -D input_rule -p tcp --dport 80 -j ACCEPT -m comment --comment "ACME"
+		ip6tables -D input_rule -p tcp --dport 80 -j ACCEPT -m comment --comment "ACME"
+	fi
+}
+
+load_options() {
+	local section="$1"
+
+	# compatibility for old option name
+	config_get_bool use_staging "$section" staging
+	if [ -z "$staging" ]; then
+		config_get_bool staging "$section" staging 0
+	fi
+	export staging
+	config_get calias "$section" calias
+	export calias
+	config_get dalias "$section" dalias
+	export dalias
+	config_get domains "$section" domains
+	export domains
+	config_get keylength "$section" keylength ec-256
+	export keylength
+	config_get dns "$section" dns
+	export dns
+	config_get acme_server "$section" acme_server
+	export acme_server
+	config_get days "$section" days
+	export days
+	config_get standalone "$section" standalone 0
```

Comment:
There is nothing to be compatible with. `webroot` is now a predefined path, so the old path is not used anymore.

Reply:
Exactly. So if someone is upgrading with an old config that sets a custom webhook, certificate renewal will now fail. We can change the default and encourage using the new functionality (so a deprecation warning is fine), but no straight-up breaking old configs.

Reply:
Maybe there is some misunderstanding? `webroot` is where a httpd gets the challenge token, which is generated at runtime. Ignoring a previously custom `webroot` shouldn't break renewal, since the new standardized location will work.